client/my-sites/navigation/navigation.jsx

```diff
@@ -2,46 +2,60 @@
  * External dependencies
  */
 import React from 'react';
+import { connect } from 'react-redux';
 
 /**
  * Internal dependencies
  */
 import observe from 'lib/mixins/data-observe';
 import SitePicker from 'my-sites/picker';
 import Sidebar from 'my-sites/sidebar';
+import { getCurrentUser } from 'state/current-user/selectors';
+import { getCurrentLayoutFocus } from 'state/ui/layout-focus/selectors';
+import { setNextLayoutFocus, setLayoutFocus } from 'state/ui/layout-focus/actions';
 
-module.exports = React.createClass( {
+const MySitesNavigation = React.createClass( {
 	displayName: 'MySitesNavigation',
 
-	mixins: [ observe( 'sites', 'user', 'layoutFocus' ) ],
+	mixins: [ observe( 'sites' ) ],
 
 	preventPickerDefault( event ) {
 		event.preventDefault();
 		event.stopPropagation();
 	},
 
 	render() {
-		var layoutFocus = this.props.layoutFocus;
-
 		return (
 			<div className="sites-navigation">
 				<SitePicker
-					layoutFocus={ layoutFocus }
+					setNextLayoutFocus={ this.props.setNextLayoutFocus }
+					setLayoutFocus={ this.props.setLayoutFocus }
+					currentLayoutFocus={ this.props.currentLayoutFocus }
 					sites={ this.props.sites }
 					allSitesPath={ this.props.allSitesPath }
 					siteBasePath={ this.props.siteBasePath }
-					user={ this.props.user }
 					onClose={ this.preventPickerDefault }
 				/>
 				<Sidebar
-					layoutFocus={ layoutFocus }
+					setNextLayoutFocus={ this.props.setNextLayoutFocus }
+					setLayoutFocus={ this.props.setLayoutFocus }
 					sites={ this.props.sites }
 					allSitesPath={ this.props.allSitesPath }
 					path={ this.props.path }
 					siteBasePath={ this.props.siteBasePath }
-					user={ this.props.user }
```

Comment:
Missed a `this.props.user` reference in `<Sidebar />` causing an error:

https://github.com/Automattic/wp-calypso/blob/f3309ee/client/my-sites/sidebar/sidebar.jsx#L658

>sidebar.jsx:721 Uncaught TypeError: Cannot read property 'get' of undefined